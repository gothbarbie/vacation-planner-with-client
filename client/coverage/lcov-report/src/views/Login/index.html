<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/views/Login</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> src/views/Login
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">18.18% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>6/33</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">18.75% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>3/16</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">36.36% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>4/11</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">18.75% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>6/32</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="Login.js"><a href="Login.js.html">Login.js</a></td>
	<td data-value="20.83" class="pic low"><div class="chart"><div class="cover-fill" style="width: 20%;"></div><div class="cover-empty" style="width:80%;"></div></div></td>
	<td data-value="20.83" class="pct low">20.83%</td>
	<td data-value="24" class="abs low">5/24</td>
	<td data-value="18.75" class="pct low">18.75%</td>
	<td data-value="16" class="abs low">3/16</td>
	<td data-value="44.44" class="pct low">44.44%</td>
	<td data-value="9" class="abs low">4/9</td>
	<td data-value="20.83" class="pct low">20.83%</td>
	<td data-value="24" class="abs low">5/24</td>
	</tr>

<tr>
	<td class="file high" data-value="index.js"><a href="index.js.html">index.js</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	</tr>

<tr>
	<td class="file low" data-value="loginActions.js"><a href="loginActions.js.html">loginActions.js</a></td>
	<td data-value="11.11" class="pic low"><div class="chart"><div class="cover-fill" style="width: 11%;"></div><div class="cover-empty" style="width:89%;"></div></div></td>
	<td data-value="11.11" class="pct low">11.11%</td>
	<td data-value="9" class="abs low">1/9</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="12.5" class="pct low">12.5%</td>
	<td data-value="8" class="abs low">1/8</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Mar 11 2018 12:53:11 GMT+0100 (CET)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
