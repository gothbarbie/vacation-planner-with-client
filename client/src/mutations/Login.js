import gql from 'graphql-tag'

export default gql`
  mutation {
    login(email: "info@hannasoderstrom.com", password: "password"){
      id
      email
    }
  }
`
