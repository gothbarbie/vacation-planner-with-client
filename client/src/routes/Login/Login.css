.login__columns {
  display: flex;
  justify-content: space-around;
}

.login__or {
  font-family: 'Avenir Next', sans-serif;
}

.login__google {
  display: flex;
  justify-content: center;
  align-items: center;
}
