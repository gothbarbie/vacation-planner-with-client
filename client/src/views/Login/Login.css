.login {
  width: 100%;
}

.login__inner {
  display: grid;
  grid-template-columns: 60% auto 30%;
  grid-template-rows: auto;
  justify-content: space-between;
  align-items: center;
}

.login__email {
  grid-column: 1 / 1;
}

.login__columns {
  /* display: grid;
  grid-template-columns: 1fr 1fr;
  grid-gap: 1rem 5rem; */
  display: flex;
  justify-content: space-around;
}

.login__or {
  font-family: 'Avenir Next', sans-serif;
}

.login__google {
  display: flex;
  justify-content: center;
  align-items: center;
}
