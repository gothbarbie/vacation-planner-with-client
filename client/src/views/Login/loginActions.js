import axios from 'axios'

import { FETCH_USER } from '../../actions/types'

export const loginUser = ({ email, password }) => async dispatch => {
  const res = await axios.post('/auth/login', {
    email,
    password,
  })

  dispatch({
    type: FETCH_USER,
    payload: res.data,
  })
}
