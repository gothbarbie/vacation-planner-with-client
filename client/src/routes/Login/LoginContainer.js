import React from 'react'

import Login from './Login'

import type { RouterHistory } from 'react-router-dom'

type Props = {
  history: RouterHistory,
}

export const LoginContainer = ({ history }: Props) => (
  <Login history={history} />
)
  
export default LoginContainer
