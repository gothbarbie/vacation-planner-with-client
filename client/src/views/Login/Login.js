// @flow

import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import { withRouter } from 'react-router-dom'
import { graphql, compose } from 'react-apollo'

import query from '../../queries/Auth'
import mutation from '../../mutations/Login'

import type { RouterHistory } from 'react-router-dom'

import PageWrapper from '../../components/PageWrapper'
import ThreeColumnsWrapper from '../../components/ThreeColumnsWrapper'
import H1 from '../../components/H1'
import Notice from '../../components/Notice'
import Form from '../../components/Form'
import Input from '../../components/FormInput'
import './Login.css'
import ButtonLink from '../../components/ButtonLink'
import Icon from '../../components/Icon'

type Props = {
  data: {
    auth: {} | void,
  },
  mutate: Function,
  history: RouterHistory,
}

type State = {
  email: {
    value: string,
    touched: boolean,
  },
  password: {
    value: string,
    touched: boolean,
  },
  errors: {
    email: string,
    password: string,
  },
}

export class Login extends Component<Props, State> {
  state = {
    email: {
      value: '',
      touched: false,
    },
    password: {
      value: '',
      touched: false,
    },
    errors: {
      email: '',
      password: '',
    },
    loginErrors: [],
  }

  componentWillUpdate (nextProps: Props) {
    if (!this.props.data.auth && nextProps.data.auth) {
      this.props.history.push('/schedule')
    }
  }

  renderAlreadyLoggedIn () {
    if (this.props.data.auth) {
      return (
        <Notice warning>
          You are already logged in, do you want to log in as another user?{' '}
          <Link to="/api/logout">Log out</Link>
        </Notice>
      )
    }
  }

  enableSubmit () {
    const { email, password, errors } = this.state

    const noErrors = !errors.email.length && !errors.password.length
    const allHasValues = email.value.length && password.value.length

    return noErrors && allHasValues
  }

  handleSubmit = (event: SyntheticEvent<any>) => {
    event.preventDefault()

    this.props
      .mutate({
        variables: {
          email: this.state.email.value,
          password: this.state.password.value,
        },
        refetchQueries: [{ query }],
      })
      .catch(res => {
        const errors = res.graphQLErrors.map(error => error.message)
        this.setState(() => { loginErrors: errors })
      })
  }

  handleChange = (event: SyntheticInputEvent<any>) => {
    if (event && event.target) {
      const target = event.target
      const value = target.type === 'checkbox' ? target.checked : target.value
      const name = target.name
      this.setState({ [name]: { ...this.state[name], value } })
    }
  }

  handleBlur = (field: string) => (event: SyntheticFocusEvent<any>) => {
    this.setState({
      [field]: { ...this.state[field], touched: true },
    })
    this.validateForm()
  }

  validateForm () {
    const email = this.state.email.value
    const password = this.state.password.value
    const errors = {}

    errors.email = !email.length && 'Email is required'
    errors.password = !password.length && 'Password is required'

    this.setState({
      errors: { ...this.state.errors, errors },
    })
  }

  showErrors = (field: string) => {
    if (this.state.errors[field] && this.state[field].touched) {
      return this.state.errors[field]
    }
  }

  render () {
    return (
      <PageWrapper>
        {this.renderAlreadyLoggedIn()}
        <H1>Login</H1>
        <ThreeColumnsWrapper>
          <div className="login__email">
            <Notice>
              First time here? <Link to="register"> Please register</Link>{' '}
              first...
            </Notice>
            <Form
              enableSubmit={this.enableSubmit()}
              onSubmit={this.handleSubmit}
              title="Enter your credentials"
            >
              <div className="login__columns">
                <Input
                  error={this.showErrors('email')}
                  handleBlur={this.handleBlur('email')}
                  handleChange={this.handleChange}
                  label="E-mail"
                  name="email"
                  type="email"
                  value={this.state.email.value}
                />
                <Input
                  error={this.showErrors('password')}
                  handleBlur={this.handleBlur('password')}
                  handleChange={this.handleChange}
                  label="Password"
                  name="password"
                  type="password"
                  value={this.state.password.value}
                />
              </div>
            </Form>
          </div>
          <div className="login__or">Or...</div>
          <div className="login__google">
            <ButtonLink secondary url="/auth/google">
              <Icon isBrand marginRight name="google-plus" />Login with Google
            </ButtonLink>
          </div>
        </ThreeColumnsWrapper>
      </PageWrapper>
    )
  }
}

export default compose(graphql(mutation), graphql(query))(withRouter(Login))
