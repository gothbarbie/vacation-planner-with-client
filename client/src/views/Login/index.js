import React from 'react'
import { Link } from 'react-router-dom'

import H1 from '../../components/common/H1'
import Notice from '../../components/common/Notice'
import Form from '../../components/common/Form'
import Input from '../../components/common/Form/Input'
import './Login.css'

export default () => (
  <section className="login">
    <H1>Login</H1>
    <div className="login__inner">
      <Notice>
        First time here? <Link to="register"> Please register</Link> first...
      </Notice>
      <Form title="Enter your credentials">
        <div className="login__columns">
          <Input 
            error="This field is required" label="E-mail" 
            name="email" />
          <Input 
            label="Password" 
            name="password" />
        </div>
      </Form>
    </div>
  </section>
)
