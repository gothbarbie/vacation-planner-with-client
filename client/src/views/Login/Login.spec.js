import React from 'react'
import { shallow } from 'enzyme'

import { Login } from './Login'

describe('views/Login', () => {
  const state = {
    email: {
      value: '',
      touched: false,
    },
    password: {
      value: '',
      touched: false,
    },
    errors: {
      email: '',
      password: '',
    },
  }

  const props = {
    data: { auth: null },
    loginUser: jest.fn(),
    history: {},
  }

  const login = shallow(<Login {...props} {...state} />)

  it('renders', () => {
    expect(login).toMatchSnapshot()
  })

  describe('#showErrors', () => {
    xit('returns errors for named field', () => {
      expect(login.showErrors('email')).toEqual()
    })
  })

})
