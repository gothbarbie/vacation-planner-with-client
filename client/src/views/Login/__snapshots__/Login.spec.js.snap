// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`views/Login renders 1`] = `
<section
  className="login"
>
  <H1>
    Login
  </H1>
  <div
    className="login__inner"
  >
    <div
      className="login__email"
    >
      <Notice>
        First time here? 
        <Link
          replace={false}
          to="register"
        >
           Please register
        </Link>
         
        first...
      </Notice>
      <Component
        onSubmit={[Function]}
        title="Enter your credentials"
      >
        <div
          className="login__columns"
        >
          <Input
            handleBlur={[Function]}
            handleChange={[Function]}
            label="E-mail"
            name="email"
            type="email"
            value=""
          />
          <Input
            handleBlur={[Function]}
            handleChange={[Function]}
            label="Password"
            name="password"
            type="password"
            value=""
          />
        </div>
      </Component>
    </div>
    <div
      className="login__or"
    >
      Or...
    </div>
    <div
      className="login__google"
    >
      <Component
        secondary={true}
        url="/auth/google"
      >
        <Icon
          isBrand={true}
          marginRight={true}
          name="google-plus"
        />
        Login with Google
      </Component>
    </div>
  </div>
</section>
`;
